File: library/src/main/java/com/pengrad/telegrambot/model/request/InlineQueryResultCachedGif.java
package com.pengrad.telegrambot.model.request;

import java.io.Serializable;

/**
 * Stas Parshin
 * 06 May 2016
 */
public class InlineQueryResultCachedGif extends InlineQueryResult<InlineQueryResultCachedGif> implements Serializable {
    private final static long serialVersionUID = 0L;

    private String gif_file_id;

    private String title;
    private String caption;
    private String parse_mode;

    public InlineQueryResultCachedGif(String id, String fileId) {
        super("gif", id);
        this.gif_file_id = fileId;
    }

    public InlineQueryResultCachedGif title(String title) {
        this.title = title;
        return this;
    }

    public InlineQueryResultCachedGif caption(String caption) {
        this.caption = caption;
        return this;
    }

    public InlineQueryResultCachedGif parseMode(ParseMode parseMode) {
        this.parse_mode = parseMode.name();
        return this;
    }
}

File: core/src/main/kotlin/com/pengrad/telegrambot/model/request/InlineQueryResultCachedGif.kt
package com.pengrad.telegrambot.model.request

import java.io.Serializable

/**
 * Stas Parshin
 * 06 May 2016
 */
class InlineQueryResultCachedGif(id: String, private val gif_file_id: String) : InlineQueryResult<InlineQueryResultCachedGif>("gif", id), Serializable {

    private var title: String? = null
    private var caption: String? = null
    private var parse_mode: String? = null

    fun title(title: String): InlineQueryResultCachedGif {
        this.title = title
        return this
    }

    fun caption(caption: String): InlineQueryResultCachedGif {
        this.caption = caption
        return this
    }

    fun parseMode(parseMode: ParseMode): InlineQueryResultCachedGif {
        this.parse_mode = parseMode.name
        return this
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}
